src/main/java/org/kohsuke/github/GHRepositoryRule.java

```diff
@@ -0,0 +1,692 @@
+package org.kohsuke.github;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Represents a repository rule.
+ */
+public class GHRepositoryRule {
+    private Type type;
+    private RulesetSourceType ruleset_source_type;
+    private String ruleset_source;
+    private long ruleset_id;
```

Comment:
Java uses camelCase, Jackson automatically translates from the snake_case JSON names.
```suggestion
    private Type type;
    private RulesetSourceType rulesetSourceType;
    private String rulesetSource;
    private long rulesetId;
```